gnovm/pkg/gnolang/alloc.go

```diff
@@ -342,34 +367,42 @@ func (alloc *Allocator) NewHeapItem(tv TypedValue) *HeapItemValue {
 }
 
 // -----------------------------------------------
+// Utilities for obtaining shallow size
 
+// keep sync with copyValueWithRefs().
 func (pv *PackageValue) GetShallowSize() int64 {
-	var (
-		allocFNames     int64
-		allocFBlocks    int64
-		allocFBlocksMap int64
-	)
-
-	for _, name := range pv.FNames {
-		allocFNames += int64(len(name)) + _allocName // string is counted as shallow size.
+	// .uverse is preloaded
+	if pv.PkgPath == ".uverse" {
+		return 0
 	}
 
-	allocFBlocks = int64(len(pv.FBlocks)) * _allocValue
+	return allocPackage
+}
 
-	for name := range pv.fBlocksMap {
-		allocFBlocksMap += int64(len(name)) + _allocName // key
-		allocFBlocksMap += _allocPointer                 // *Block
```

Comment:
Remove this for now. as these values were not allocated (of course the take up memory), so not count here either. will improve in future.

Reply:
1. No the check logic you mentioned is not included in this PR, it will be shipped with the 3 take in my plan;
2. yes, that it is.

Reply:
So this PR fills the gap between allocation and GC counting as much as possible, but it hasn’t fully achieved the newBytes <= oldBytes comparison yet, and there are still some minor details to address, like the second one, and `StringValue` counting, etc.